source/handbook/cpp-coding-guidelines.rst

```diff
@@ -1,289 +1,238 @@
-
-     for (size_t j = 0;j < sections.size();++j)
+Arrays
+-------
 
-* Always put a space after coma ',' in function args, like:
+If you pass size of an array somewhere, always give the size variable a name, that suggests what it actually is:
 
-  .. code-block:: c++
+1. So if you actually expect **number of elements**, use name with ``Count`` postfix [e.g. ``nodeCount``]
 
-     outputAsciiString(buffer, something, elsewhere);
+2. If you want number of **bytes** use name like ``size_t nameSize``
 
-  Not:
+Includes
+********
 
-  .. code-block:: c++
+1. Always use "/" in includes and NEVER "\\", (C standard WG14/N1256 point 6.4.7, C++ standard ISO/IEC 14882:2003 point 2.8, C++ standard ISO/IEC 14882:2011 (from working draft N3225) point 2.9)
 
-     outputAsciiString(buffer,something,elsewhere);
+2. Try to include the minimal number of files required for a successful build. Corollary: don't include anything you don't need.
 
-* Do NOT leave whitespaces at line-endings (here's a regex for "Quick Replace" in visual studio: ``[ ]+$``)
+3. Order includes from most specific to most general.
 
-Types
+Style
 *****
 
-* ``size_t`` should be used whenever dealing with data size (in many cases ``auto`` is fine too):
-
-  * The result of ``sizeof()`` is ``size_t``
-
-  * Difference between pointer types is always ``size_t``
-
-  * Index of an element in an array should be of ``size_t`` type
-
-  * The result of ``strlen()`` should usually be ``size_t``
-
-  * Most STL containers uses ``size_t`` as default size, count, length and index type
-
-* Please use types defined in **stdint.h** (``uint8_t``, ``uint16_t``, ``uint32_t``, etc.)
-
-Disputable:
-
-* Please avoid using **signed** types and **signed** math unless it's really necessary and reasonable.
-
 Classes, Methods and Members
-****************************
+----------------------------
 
-* Classes should be named using ``CamelCase`` (first letter capital)
+Recommended class order:
 
-* Class order (disputable):
+1. Private constants (as they are usually used early)
 
-  1. Private constants (as they are usually used early)
+2. Public constants
 
-  2. Public constants
+3. Methods
 
-  3. Methods (if possible public, protected, private)
+   1. Constructors
 
-  4. Fields
+   2. Instance methods
 
-     1. Public members (should probably be used only for POD types)
+   3. Static methods
 
-     2. Protected members
+4. Fields
 
-     3. Private members
+   1. Public members (should probably be used only for POD types)
 
-Arguments
-*********
+   2. Protected members
 
-* Avoid passing arguments as pointers (reference is always preferred) unless it's really intended and needed.
+   3. Private members
 
-* Use ``const`` references or ``const`` types when possible.
+Use an access modifier (``public:``, ``private:``) for each section even if the modifier is redundant.
 
 Special Names
-*************
+-------------
 
 * BlockChain not Blockchain
 
 * Timestamp not TimeStamp
 
 * Filesystem not FileSystem
```

Comment:
I think there are more, but we have it in linter and haven't been updating this list...

Reply:
Please not "BlockChain". 
It is Blockchain. As defined everywhere else in the world. 😅